language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -394,22 +401,21 @@ applyStagedBuiltinName
     :: (GShow uni, GEq uni, DefaultUni <: uni, Closed uni, uni `Everywhere` ExMemoryUsage)
     => StagedBuiltinName
     -> [WithMemory Value uni]
-    -> CekM uni (ConstAppResult uni ExMemory)
+    -> CekM uni (ConstAppResult (WithMemory Term uni))
 applyStagedBuiltinName n@(DynamicStagedBuiltinName name) args = do
     DynamicBuiltinNameMeaning sch x exX <- lookupDynamicBuiltinName name
     applyTypeSchemed n sch x exX args
-applyStagedBuiltinName (StaticStagedBuiltinName name) args = do
-    params <- builtinCostParams
-    applyBuiltinName params name args
+applyStagedBuiltinName (StaticStagedBuiltinName name) args =
+    applyBuiltinName name args
```

Comment:
Moved the logic of retrieving costing parameters into `applyBuiltinName` (as it runs in a monad where we have access to the parameters).